content/en/BestPractices/defend-against-ssrf.md

```diff
@@ -0,0 +1,257 @@
+# Defend Against Server-Side Request Forgery
+## Description 
+
+[Server-Side Request Forgery (SSRF)](https://owasp.org/Top10/A10_2021-Server-Side_Request_Forgery_%28SSRF%29/) is a server-side security vulnerability that allows a malicious actor (attacker) to make arbitrary requests from the application's server. 
+
+
+SSRF has a wide range of impacts. For example, it could allow an attacker to perform a simple external interaction to steal sensitive AWS metadata information that could eventually lead to an account takeover. SSRF, given its impact and likelihood, has been rated at 10th position in the OWASP Top 10 (2021) vulnerabilities. 
+
+
+
+[Optional] On July 29th, 2019, a widely known organization, CapitolOne, suffered a data breach that exposed 140,000 Social Security Numbers, 80,000 bank account numbers, and 1 million social insurance numbers. The initial attack vector was an SSRF exploit that exposed AWS credential keys. Read more about the attack here. 
+
+
+
+An attacker may achieve the following through a successful SSRF exploitation: 
+
+
+
+- Local/Remote Port Scan
+
+- Local File Read
+
+- Interact with internal apps/service/network
+
+- Remote Code Execution (RCE) by chaining services on the internal network
+
+- Read Metadata Cloud (AWS, Azure, Google Cloud, Digital Ocean, etc.)
+
+- Reflected Cross-Site Scripting (XSS)
+
+- Cross-Site Request Forgery (CSRF)
+
+
+
+## Examples
+
+
+
+### Scenario - 1: Simple SSRF Attack allows access to localhost
+
+
+
+```Vulnerable Code Snippet
+
+from flask import request
+
+import urllib
+
+
+
+@app.route('/proxy')
+
+def proxy():
+
+    url = request.args["url"]
+
+    return urllib.request.urlopen(url).read() 
+
+```
+
+
+
+In the above code, the application accepts the user-supplied input through the *url* parameter and passes it to the *urllib.request.urlopen(url).read()* function. SSRF attacks are possible since the malicious URLs supplied by the users are accepted and processed without filtration. 
+
+
+
+Example: https://www.target.com/test/?url=http://127.0.0.1 will allow an attacker to access the `127.0.0.1 (localhost)` by passing it to the vulnerable code. 
```

Comment:
Would it be a little clearer to say something like `For example, if an attacker submits https://www.target.com/test/?url=http://127.0.0.1, the server attempts to pass 127.0.0.1 (localhost) to the vulnerable code, then call out to that IP address`
(if I understand how SSRF works as well as I think I do)

Mostly this adds a bit of context for how the vulnerability works, as well as showing the cause-effect relationship.